fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/Client.java

```diff
@@ -463,7 +459,7 @@ public boolean isWithdrawn() {
         return ClientStatus.fromInt(this.status).isWithdrawn();
     }
 
-    public Map<String, Object> update(final JsonCommand command) {
+    public Map<String, Object> update(final ExternalId externalId, final JsonCommand command) {
```

Comment:
Please move all the "command" related logic out of the entity.

Reply:
Done, Interface and Service created `ClientDomainService`

Reply:
i thought you will move this logic into the only service which calls it... :/

Reply:
Done